math/mathcore/inc/Math/DistSampler.h

```diff
@@ -243,17 +261,14 @@ class DistSampler {
       return *fRange;
    }
 
-
-
-
 private:
 
    // private methods
 
-   bool fOwnFunc;                         // flag to indicate if the function is owned
-   mutable std::vector<double> fData;     // internal array used to cached the sample data
-   ROOT::Fit::DataRange    *   fRange;    // data range
-   const ROOT::Math::IMultiGenFunction * fFunc; // internal function (ND)
+   bool fOwnFunc;                         /// flag to indicate if the function is owned
+   mutable std::vector<double> fData;     ///! internal array used to cached the sample data
```

Comment:
Just for my understanding, what is the point of having the `fData` member marked as transient with `/!` when the DistSampler doesn't have a `ClassDef`? If there is no ClassDef, doesn't this mean that the class is not serializable and it doesn't matter if a member is transient or not?

Reply:
Ah I think I get it, it's because if you have a class deriving from it like the TUnuranSampler (which has a ClassDef introduced with this commit), it also needs to know which members of the base classes should be transient or not.

Reply:
@guitargeek  a class can be streamed in ROOT also if it has not a ClassDef. ClassDef only defines the version of the class